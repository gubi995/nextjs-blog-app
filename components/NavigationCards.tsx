import React from 'react';
import Router from 'next/router';
import { Paper, Typography, styled } from '@material-ui/core';

const StyledNavigationCards = styled('div')(({ theme }) => ({
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'space-evenly',
  margin: theme.spacing(5),
}));

const StyledPaper = styled(Paper)(({ theme }) => ({
  display: 'flex',
  flexBasis: '45%',
  justifyContent: 'center',
  padding: theme.spacing(5),
  margin: theme.spacing(2),
  textAlign: 'center',
  cursor: 'pointer',
  minWidth: '300px',
}));

const NavigationCards = () => {
  return (
    <StyledNavigationCards>
      <StyledPaper variant="outlined" onClick={() => Router.push('/repo')}>
        <Typography color="primary" variant="h6" component="h4">
          Take a look on my GitHub repos
        </Typography>
      </StyledPaper>
      <StyledPaper variant="outlined" onClick={() => Router.push('/blog')}>
        <Typography color="primary" variant="h6" component="h4">
          Find some interesting topic between my blog posts
        </Typography>
      </StyledPaper>
    </StyledNavigationCards>
  );
};

export default NavigationCards;
