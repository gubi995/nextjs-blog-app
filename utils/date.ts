export const toReadableDate = (dateTime: string) => new Date(dateTime).toLocaleDateString('en-CA');
