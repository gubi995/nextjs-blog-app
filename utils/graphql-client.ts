import { GraphQLClient } from 'graphql-request';

const client = new GraphQLClient(process.env.API_ENDPOINT!, {
  headers: {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${process.env.API_TOKEN!}`,
    Accept: 'application/json',
  },
});

export default client;
