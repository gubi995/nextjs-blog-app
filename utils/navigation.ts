import Router from 'next/router';

export const navigateAndScrollTop = async (url: string, as?: string) => {
  await Router.push(url, as);

  window.scrollTo(0, 0);
};
