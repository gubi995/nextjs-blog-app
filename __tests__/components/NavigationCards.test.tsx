import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { navigateAndScrollTop } from '../../utils/navigation';
import NavigationCards from '../../components/NavigationCards';

jest.mock('../../utils/navigation');

describe('NavigationCards', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("should rendered component matches with it's the snapshot", () => {
    const { asFragment } = render(<NavigationCards />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should navigate to /repo and scroll to top', () => {
    const { getByText } = render(<NavigationCards />);

    const paperText = getByText('Take a look on my GitHub repos');

    fireEvent.click(paperText);

    expect(navigateAndScrollTop).toHaveBeenCalledTimes(1);
    expect(navigateAndScrollTop).toHaveBeenCalledWith('/repo');
  });

  test('should navigate to /blog and scroll to top', () => {
    const { getByText } = render(<NavigationCards />);

    const paperText = getByText('Find some interesting topic between my blog posts');

    fireEvent.click(paperText);

    expect(navigateAndScrollTop).toHaveBeenCalledTimes(1);
    expect(navigateAndScrollTop).toHaveBeenCalledWith('/blog');
  });
});
