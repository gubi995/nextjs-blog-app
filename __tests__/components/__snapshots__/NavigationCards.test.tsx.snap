// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationCards should rendered component matches with it's the snapshot 1`] = `
<DocumentFragment>
  <div
    class="div-root-1 div-root-2"
  >
    <div
      class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 WithStyles(ForwardRef(Paper))-root-4 MuiPaper-outlined MuiPaper-rounded"
    >
      <h4
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
      >
        Take a look on my GitHub repos
      </h4>
    </div>
    <div
      class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 WithStyles(ForwardRef(Paper))-root-63 MuiPaper-outlined MuiPaper-rounded"
    >
      <h4
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
      >
        Find some interesting topic between my blog posts
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
